### PR TITLE
update tls

### PR DESCRIPTION
**Pull request by gitlawr**
https://github.com/harvester/harvester/issues/1131
https://github.com/harvester/harvester/issues/1193

Sync cacerts setting;
Add VIP to certificate SANs;
Add 80 port to the expose service so that http://vip is accessible(redirected to https://).

### CODE REVIEW

File: pkg/controller/master/rancher/embedded_rancher.go [hunk 1]
```diff
@@ -18,6 +20,10 @@ func (h *Handler) RancherSettingOnChange(key string, setting *rancherv3api.Setti
 		return nil, nil
 	}
 
+	if setting.Name == "internal-cacerts" && setting.Value != "" {
```
Comment (guangbochen) [inline pkg/controller/master/rancher/embedded_rancher.go]: let's move the `internal-cacerts` name to the constant at `register.go` for better consistency.

File: pkg/controller/master/rancher/embedded_rancher.go [hunk 2]
```diff
@@ -30,3 +36,67 @@ func (h *Handler) RancherSettingOnChange(key string, setting *rancherv3api.Setti
...
+// initializeTLS writes internal-cacerts to cacerts value and adds the VIP to certificate SANs
+// cacerts is used in generated kubeconfig files
+func (h *Handler) initializeTLS(setting *rancherv3api.Setting) error {
+	cacerts, err := h.RancherSettingCache.Get("cacerts")
+	if err != nil {
+		return err
+	}
+	if cacerts.Value != "" {
+		return nil
+	}
```
Comment (guangbochen) [inline pkg/controller/master/rancher/embedded_rancher.go:49]: seems duplicate with `h.setCACerts(setting.Value)`

File: pkg/controller/master/rancher/embedded_rancher.go [hunk 3]
```diff
@@ -30,3 +36,67 @@ func (h *Handler) RancherSettingOnChange(key string, setting *rancherv3api.Setti
...
+	return nil
+}
+
+func (h *Handler) setCACerts(cert string) error {
+	cacerts, err := h.RancherSettingCache.Get("cacerts")
+	if err != nil {
+		return err
+	}
+	if cacerts.Value == cert {
```
Comment (guangbochen) [inline pkg/controller/master/rancher/embedded_rancher.go:66]: should this be `if cacerts.Value == cert || cacerts.Value != ""`

Comment (gitlawr) [reply pkg/controller/master/rancher/embedded_rancher.go:66]: The `cacerts.Value != ""` check is used to skip the update when it is already initialized. But for a function like `setCACerts(cert)`, it doesn't have to care about that.

**Comment by gitlawr** [comment]
updated

**Review by guangbochen — APPROVED** [review]
LGTM

**Pull request merged**